spec/spec.md

```diff
@@ -0,0 +1,146 @@
+If the main purpose of this server is to be a proxy on top of an existing catalog service,
+converting back and forth between Lance REST API models and native API models of the catalog service,
+then this Lance catalog server is called a **Lance Catalog Adapter**.
+
+### Object Hierarchy
+
+The REST Catalog provides a container concept called **Namespace** and 
+exposes a 2 level hierarchy below the root REST catalog server to organize tables.
+A catalog contains a list of namespaces, and a namespace contains a list of tables.
+
+### Relationship with Lance Table on Storage
+
+This section defines the possible relationships between a Lance table definition in a REST catalog and
+the corresponding Lance table sitting in some storage system.
+A catalog can choose to support one or multiple types of relationships when integrating with the REST protocol.
+
+#### Catalog Managed Table
+
+A catalog managed Lance table is a table that is fully managed by a catalog.
+The catalog must contain information about the latest version of the Lance table.
+Any modifications to the table must happen through the catalog.
+If a user directly modifies the underlying table in the storage bypassing catalog,
+the catalog must not reflect the changes in the table to the catalog users.
+
+This mode ensures the catalog service is aware of all activities in the table,
+and can thus fully enforce any governance and management features for the table. 
+
+#### Storage Managed Table
+
+A storage managed Lance table is a table that is fully managed by the storage with a metadata definition in catalog.
+The catalog only contains information about the table directory location.
+It is expected that a tool finds the latest version of the Lance table based on the contents 
+in the table directory according to the Lance format specification.
+A modification to the table can happen either directly against the storage,
+or happen as a request to the catalog, where the catalog is responsible for applying the corresponding
+change to the underlying storage according to the ance format specification.
+
+This mode is more flexible for real world ML/AI workflows 
+but the catalog loses full visibility and control over the actions performed against the table,
+so it will be harder to enforce any governance and management features for storage managed tables.
+
+## Tool Integration Guidelines
+
+The following are guidelines for tools to integrate with Lance catalogs.
+Note that these are recommendations rather than hard requirements.
+The goal of these guidelines is to offer a consistent user experience across different tools.
+
+### Configurations
+
+We recommend tools to offer the following configurations in some form or shape 
+for users to configure connection to a Lance catalog:
+
+| Config Key | Description                                                                                 | Required?                   | 
```

Comment:
Do we need to describe the default value?

Reply:
if it is required, why does it need a default value?